lua/plenary/popup/init.lua

```diff
@@ -412,22 +497,10 @@ function popup.create(what, vim_options)
 
   -- callback
   if vim_options.callback then
-    popup._callbacks[bufnr] = function()
-      -- (jbyuki): Giving win_id is pointless here because it's closed right afterwards
-      -- but it might make more sense once hidden is implemented
-      local row, _ = unpack(vim.api.nvim_win_get_cursor(win_id))
-      vim_options.callback(win_id, what[row])
-      vim.api.nvim_win_close(win_id, true)
-    end
-    vim.api.nvim_buf_set_keymap(
-      bufnr,
-      "n",
-      "<CR>",
-      '<cmd>lua require"plenary.popup".execute_callback(' .. bufnr .. ")<CR>",
-      { noremap = true }
-    )
+    popup._callback_fn[win_id] = vim_options.callback
   end
 
+  -- TODO: Wonder what this is about? Debug? Convenience to get bufnr?
```

Comment:
I think what this does if I'm not mistaken @tjdevries @Conni2461 it's that checks for `vim_options.finalize_callback` to be (it could be null) and if does exists calls it, maybe there's a part of the code where it removes it or it's set to false, will look into that.

Reply:
> `vim_options.finalize_callback`

The question is about why it exists at all. It was added with #513.

@xactlyblue Could you clarify the purpose of #513? It doesn't seem to have anything to do with popups.

Reply:
Yeah why it exists, that idk, if @xactlyblue could clarify on that regard, would be nice. I tried to find where it was being used or defined and this is what I got.

I added this to log when it's called or not:

![image](https://github.com/user-attachments/assets/a85f1906-d79f-4b7b-a9ea-d33ced3ec883)

Results from `make test`:

![Screenshot 2024-10-09 at 18 04 38](https://github.com/user-attachments/assets/bdaff51d-2c89-4bd6-bc4b-c0d14c013232)

Video demo when using `Telescope` which creates a `popup`:

https://github.com/user-attachments/assets/bc89065f-cb03-4f55-9088-0beda0abacd1

Reply:
Tried both in this PR and in master btw, in case it was related to something this PR was removing.

Reply:
> if @xactlyblue could clarify on that regard

Hey. I made that change a while ago to address a problem I was having where I needed to access some information about the buffer created for the popup _(I forgot the details to be honest, but if you need them I could probably go through my dotfiles commit history to figure out what my specific use-case was at the time)_. From what I remember it just creates an easier way for the user to manually access the ``bufnr`` and ``win_id`` of the popup for whatever reason they might need it. I probably should've added a comment explaining the changes to clear up any possible confusion from it, sorry!

Reply:
> creates an easier way for the user to manually access the bufnr and win_id

Thanks, no problem. Debug convenience. Do you agree there's no reason to keep it in a final product?

Consider `nvim_win_get_buf(win_id)`

Reply:
Sorry I wasn't very clear with my wording. I'm not an active contributor to plenary, my use case was outside of any internal modifications _(I believe it had something to do with modifying the zindex of a window to prevent overlap or something along those lines somewhere in my nvim config)_. 

I'm not really familiar with the inner-workings of plenary and I couldn't find any simple, existing method to access the bufnr of a created popup and so I figured the simplest solution would be to add an optional callback to allow the user to make any necessary changes.

I looked through my NeoVim configuration and I'm not currently using it so I _personally_ don't mind if it's removed but I'm not sure if anyone else is making use of it in their own configurations _(though I'd guess the odds of that are fairly slim considering it's not really documented anywhere)_.

Reply:
I understood (I think). I didn't see a reason for it, other than convenience.

> simple, existing method to access the bufnr of a created popup

I'm new to `neovim`, I don't know what was available when. Currently `nvim_win_get_buf(win_id)` or `winbufnr()` gets the buffer number from a window. May be other ways.

I'm not asking you to remove it, just wanted to make sure there wasn't something I missed.